yourgov/components/FormMobileFoodVendorPermit/task1/FormTask1Step3.tsx

```diff
@@ -1,116 +1,73 @@
 	const { submitStep } = useFormTask1Context();
 
-	const scrollToField = useScrollToField();
-	const errorRef = useRef<HTMLDivElement>(null);
-	const [focusedError, setFocusedError] = useState(false);
-
 	const {
 		watch,
 		register,
 		handleSubmit,
 		formState: { errors },
 	} = useForm<Task1Step3FormSchema>({
 		defaultValues: formState.task1?.step3,
-		resolver: yupResolver(task1Step3FormSchema),
+		resolver: isSavingBeforeExiting
+			? undefined
+			: zodResolver(task1Step3FormSchema),
+		mode: 'onSubmit',
+		reValidateMode: 'onBlur',
 	});
 
 	const onSubmit: SubmitHandler<Task1Step3FormSchema> = async (data) => {
-		setFocusedError(false);
+		if (isSavingBeforeExiting) {
+			return;
+		}
 		await submitStep();
 		setFormState({
 			...formState,
-			task1: { ...formState.task1, step3: { ...data, completed: true } },
+			task1: {
+				...formState.task1,
+				step3: {
+					...data,
+					completed: !isSavingBeforeExiting,
+					started: true,
+				},
+			},
 		});
 	};
 
-	const onError = () => {
-		setFocusedError(false);
-	};
-
-	// As our form schema contains nested objects, we are converting the errors from a nested object to a simple flat array
-	const flatErrors = Object.entries(errors)
-		.map(([key, value]) => {
-			if ('message' in value) return { key, message: value.message };
-		})
-		.filter((item): item is { key: string; message: string } =>
-			Boolean(item?.message)
-		);
-
-	// Only show the page alert if there is more than 1 error
-	const hasErrors = Object.keys(errors).length > 1;
+	const showErrorAlert = hasMultipleErrors(errors);
 
-	useEffect(() => {
-		if (hasErrors && !focusedError) {
-			errorRef.current?.focus();
-			setFocusedError(true);
-		}
-	}, [hasErrors, focusedError, errors]);
-
-	const isPostalAddressSameAsStreetAddress = watch(
-		'isPostalAddressSameAsStreetAddress'
+	const isPostalAddressSameAsBusinessAddress = watch(
+		'isPostalAddressSameAsBusinessAddress'
 	);
 
 	return (
 		<FormTask1Container
 			formTitle="Business address"
 			formIntroduction="Add your business address."
-			formCallToAction={<FormRequiredFieldsMessage />}
 		>
-			<Stack
-				as="form"
-				gap={3}
-				onSubmit={handleSubmit(onSubmit, onError)}
-				noValidate
-			>
-				{hasErrors && (
-					<PageAlert
-						ref={errorRef}
-						tabIndex={-1}
-						tone="error"
-						title={<PageAlertTitle as="h2">There is a problem</PageAlertTitle>}
-					>
-						<Text as="p">
-							Please correct the following fields and try again
-						</Text>
-						<UnorderedList>
-							{flatErrors.map((error) => (
-								<ListItem key={error.key}>
-									<TextLink href={`#${error.key}`} onClick={scrollToField}>
-										{error.message}
-									</TextLink>
-								</ListItem>
-							))}
-						</UnorderedList>
-					</PageAlert>
-				)}
+			<Stack as="form" gap={3} onSubmit={handleSubmit(onSubmit)} noValidate>
+				{showErrorAlert && <FormPageAlert errors={errors} />}
```

Comment:
I feel like the `PageAlert` shouldn't update until the form is submitted again. It can take the focus unexpectedly, and makes the UI jump around on each errored field fix. E.g. if you:

1. Load the page
2. Press "Save and continue"
3. Fill out the visible form fields
4. Uncheck the "Same as business address" checkbox
5. Tab through newly rendered form fields
6. PageAlert renders again after two errored fields and steals focus + the page is scrolled

Reply:
Also, if you have errors in your "Postal address" fields but re-check the "Same as business address" checkbox then the Page Alert persists. You can save and continue, though. This is a different UX than just filling out the form and having the Page Alert update as you fill out fields.

Reply:
Yeah, this `reValidateMode: 'onBlur'` is not good

Reply:
A separate PR will change to "onChange" and make all the relevant fixes to page alerts etc